### PR TITLE
Handle urban area with missing daily VMT per capita

### PR DESCRIPTION
**Pull request by rouille**
[Pull Request doc](https://breakthrough-energy.github.io/docs/user/git_guide.html#d-pull-request)

### Purpose
Use annual VMT per capita in state to calculate the total annual VMT in urban areas that have missing daily VMT per capita

### What the code is doing
* Remove rows where indices are set to `[no data]` in place of a urban area (single row)
* Use a `if`/`else` statement when calculating the total annual VMT in urban areas

### Testing
Manual testing to check that some urban areas with missing data have reasonable values compared to other urban areas in the state.

### Where to look
* It's helpful to clarify where your new code lives if you moved files around or there could be confusion/

* What files are most important?

### Usage Example/Visuals
```
>>> from prereise.gather.demanddata.transportation_electrification.generate_scaling_factors import (
...     census_ua_url,
...     census_state_url,
...     tht_data_url,
...     load_census_ua,
...     load_census_state,
...     load_dot_vmt_per_capita,
...     calculate_vmt_for_ua,
...     calculate_vmt_for_state,
...     calculate_urban_rural_fraction,
...     get_efs_vmt_projection_for_state,
...     generate_scaling_factor,
... )
>>> from prereise.gather.demanddata.nrel_efs.get_efs_annual_data import (
...     get_efs_annual_data,
...     nrel_annual_efs_url,
... )
>>> 
>>> census_ua = load_census_ua(census_ua_url)
>>> census_state = load_census_state(census_state_url)
NST01
>>> tht_ua, tht_state = load_dot_vmt_per_capita(tht_data_url)
>>> vmt_ua = calculate_vmt_for_ua(census_ua, tht_ua, tht_state)
>>> vmt_ua["OR"]
                      Annual VMT
Portland, OR-WA     1.024164e+10
Salem, OR           1.485755e+09
Albany, OR          4.917825e+08
Corvallis, OR       2.872120e+08
Medford, OR         1.002208e+09
Eugene, OR          1.583623e+09
Grants Pass, OR     4.358975e+08
Bend, OR            4.988977e+08
Walla Walla, WA-OR  7.614401e+07
Longview, WA-OR     1.876433e+07
```

### Time estimate
5min

### CODE REVIEW

**Review by BainanXia — APPROVED** [review]
Looks good. Thanks

File: prereise/gather/demanddata/transportation_electrification/generate_scaling_factors.py [hunk 1]
```diff
@@ -129,17 +129,19 @@ def load_dot_vmt_per_capita(path):
...
         df_state.rename(index=state2abv).squeeze(),
     )
 
 
-def calculate_vmt_for_ua(census_ua, tht_ua):
+def calculate_vmt_for_ua(census_ua, tht_ua, tht_state):
     """Calculate the total annual Vehicle Miles Traveled (VMT) in urban areas
 
     :param dict census_ua: dictionary as returned by :func:`load_census_ua`
     :param pandas.Series tht_ua: vmt per capita in urban areas as returned by
```
Comment (danlivengood) [inline prereise/gather/demanddata/transportation_electrification/generate_scaling_factors.py]: Given our own misremembering, I'd suggest
`param pandas.Series tht_ua: daily vmt per capita in urban areas`

File: prereise/gather/demanddata/transportation_electrification/generate_scaling_factors.py [hunk 2]
```diff
@@ -129,17 +129,19 @@ def load_dot_vmt_per_capita(path):
...
         :func:`load_dot_vmt_per_capita`
+    :param pandas.Series tht_state: vmt per capita in state as returned by
```
Comment (danlivengood) [inline prereise/gather/demanddata/transportation_electrification/generate_scaling_factors.py]: Same reasoning:
`:param pandas.Series tht_state: annual vmt per capita in state`

File: prereise/gather/demanddata/transportation_electrification/generate_scaling_factors.py (hunk 1 above)
Comment (rouille) [reply prereise/gather/demanddata/transportation_electrification/generate_scaling_factors.py]: Done

File: prereise/gather/demanddata/transportation_electrification/generate_scaling_factors.py (hunk 2 above)
Comment (rouille) [reply prereise/gather/demanddata/transportation_electrification/generate_scaling_factors.py]: Done

**Review by danlivengood — APPROVED** [review]
Looks good to me.  Thanks!

**Pull request merged**